tools/topsites.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import datetime
+import hashlib
+import hmac
+import os
+import sys
+from urllib import urlencode
+from xml.dom.minidom import parseString
+
+import requests
+from requests.exceptions import ConnectionError
+from sqlalchemy import create_engine
+from sqlalchemy import Column
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+# Add webcompat module to import path
+sys.path.append(os.path.realpath(os.pardir))
+from webcompat import app
+
+# Constant for alexa top site API
+ATS_ACTION_NAME = 'TopSites'
+ATS_RESPONSE_GROUP_NAME = 'Country'
+ATS_SERVICE_HOST = 'ats.amazonaws.com'
+ATS_AWS_BASE_URL = 'http://' + ATS_SERVICE_HOST + '/?'
+ATS_DATEFORMAT = '%Y-%m-%dT%H:%M:%S.%f'
+ATS_HASH_ALGORITHM = 'HmacSHA256'
+ATS_COUNT = 100
+
+# Regions to dump to topsites.db
+REGIONS = ['GLOBAL', 'US', 'FR', 'IN', 'DE', 'TW', 'ID', 'HK', 'SG', 'PL',
+           'GB', 'RU']
+
+# Alexa top site access and secret key
+ats_access_key = None
+ats_secret_key = None
+
+
+# Cache parsed sites, change priority if raised
+topsites = {}
+
+engine = create_engine('sqlite:///' + os.path.join(
+    app.config['BASE_DIR'], 'topsites.db'))
```

Comment:
@MDTsai Following our discussion in San Francisco. Maybe we should rename this `topsites-new.db`. So that when we run it, we don't crush the previous one and the fabric file or a script will be in charge of rotating the DB once the db has been populated. 

Once this is done. Feel free to merge if I'm already off.